### PR TITLE
#233 Save song as favourite

### LINKED ISSUE #233: Save song as favourite
#### 

What do you think if we implement a feature to save a song as favourite? Having another list that will show the favourite ones or even in the same list but with a button to filter the favourites.

- [ ] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This issue is for enhancement/bug fixing.
- [ ] This issue is an addition to JSON database.
- [ ] I have gone through [Contribution Guidelines](https://github.com/anshumanv/ongaku/blob/master/CONTRIBUTING.md).
- [x] Assign this issue to me.

### PR DESCRIPTION
**Pull request by OuKevin**
Fixes #233 

Added a star icon to the top right of the screen that will store a song in local storage.
Added a preference button that can be used to filter song list to display only favorite songs.

- [X ] This issue is for enhancement/bug fixing.
- [ ] This issue is an addition to JSON database.
	- [ ] Pull Request Title is of form `Added [Anime Name]`.
- [ ] This addition is already on ongaku.
	- [ ] This PR is an improvement to current addition.
- [X ] I have checked my addition locally and it is working and won't break existing entries.
- [X ] I have gone through [Contribution Guidelines](https://github.com/anshumanv/ongaku/blob/master/CONTRIBUTING.md).
- [X ] This pull request will be closed if I failed to update it even once in a week.

### CODE REVIEW

**Comment by anshumanv** [comment]
@OuKevin Awesome Work! :smile: 
Thanks for your contributions. :+1:

File: js/ongaku.js [hunk 1]
```diff
@@ -283,11 +296,42 @@
...
+			displayStar()
+		} else {
+			localStorage.setItem(order[0].name, true)
+			displayStar()
+		}
+	}
+
+	//displays a filled in star if in favorite list
+	function displayStar() {
+		var element = document.getElementById("starImg");
```
Comment (anshumanv) [inline js/ongaku.js]: You have already grabbed the element in line 16, please use that instead

File: js/ongaku.js [hunk 2]
```diff
@@ -247,6 +248,16 @@
...
 	});
 
+	//filters list to only display favorites if checked
+	function filteredByFavorites(songList, isChecked) {
+		if ( isChecked ) {
+			return songList.filter( function(song) {
+				return localStorage.getItem(song.name)
+			});
+		}
+		return songList
```
Comment (anshumanv) [inline js/ongaku.js]: Missed `;`

File: js/ongaku.js [hunk 3]
```diff
@@ -283,11 +296,42 @@
 		}
 	}
 
+	//function to check if song is a favorite
+	function isFavorite() { 
+		return localStorage.getItem([order[0].name])
```
Comment (anshumanv) [inline js/ongaku.js]: Please add `;`

File: js/ongaku.js [hunk 4]
```diff
@@ -283,11 +296,42 @@
...
+	}
+
+	//function to handle favorites in local storage
+	function toggleFavorites() {
```
Comment (anshumanv) [inline js/ongaku.js]: Add `;` wherever applicable. :smile:

File: index.html [hunk 5]
```diff
@@ -56,6 +56,13 @@
 							<span class="round-btn"></span>
 						</div>
 					</div>
+					<div class="ost">
```
Comment (anshumanv) [inline index.html]: Why class `ost`?

File: js/ongaku.js [hunk 6]
```diff
@@ -283,11 +296,42 @@
...
+		if( isFavorite() ) {
+			localStorage.removeItem(order[0].name)
+			displayStar()
+		} else {
+			localStorage.setItem(order[0].name, true)
+			displayStar()
+		}
+	}
+
+	//displays a filled in star if in favorite list
```
Comment (anshumanv) [inline js/ongaku.js]: Improve this docstring please

**Comment by OuKevin** [comment]
No problem! Cool project 👍

**Review by anshumanv — APPROVED** [review]
:+1:

**Comment by anshumanv** [comment]
Thanks for the quick fix. :smile:

**Pull request merged**